### PR TITLE
fix(ctb): Package pre-pr Target Fixes

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Following the removal of the `storage-snapshot` target in #8303, the `pre-pr` target needed to be updated to use the new `snapshots` target.

This pr updates `pre-pr` to use `snapshots` and also changes the go bindings building to use the `no-build` target.

## Summary by CodeRabbit

- **Refactor**
  - Streamlined the snapshot verification process to enhance the reliability of build checks.

- **Chores**
  - Updated scripts to enforce stricter execution standards, ensuring consistency in snapshot generation and comparison.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #8363** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8363?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8363?utm_source=stack-comment).

**Review by clabby — APPROVED** [review]
Nice

File: .circleci/config.yml [hunk 1]
```diff
@@ -432,12 +432,7 @@ jobs:
       - run:
           name: snapshots
           command: |
-            pnpm snapshots
-            if git diff --exit-code snapshots > /dev/null; then
-              [ -z "$(git ls-files --others --exclude-standard snapshots)" ] || echo "export SNAPSHOTS_STATUS=1" >> "$BASH_ENV"
-            else
-              echo "export SNAPSHOTS_STATUS=1" >> "$BASH_ENV"
```
Comment (trianglesphere) [inline .circleci/config.yml:439]: This is change to how we do error handling in this job. Previously we would set error codes for each job & only fail after each job had been completed. Now we will do that for some steps except for this step which will fail immediately.

Comment (refcell) [reply .circleci/config.yml:439]: Ah, i have an uncommitted change locally to add the else clause here to set the snapshots status variable. Will commit right now

**Review by tynes — APPROVED** [review]

**Pull request merged**